tests/karate/src/test/java/common/net/MessageFilters.java

```diff
@@ -0,0 +1,29 @@
+package common.net;
+
+import java.util.function.Predicate;
+
+/**
+ * This class contains useful message filters that can be used when using :
+ *  - {@link MessageBuffer#peek(Predicate)}
+ *  - {@link MessageBuffer#peekAll(Predicate)}
+ *  - {@link MessageBuffer#take(Predicate)}
+ *  - {@link MessageBuffer#takeAll(Predicate)}
+ *  - {@link MessageBuffer#takeTimeout(Predicate, long)}
+ */
+public final class MessageFilters {
+
+  private MessageFilters() {
+    // Static lass, this should never be called
+  }
+
+
+  /**
+   * Only accept the messages with the given method
+   *
+   * @param method to accept
+   * @return true if the message is accepted
+   */
+  public static Predicate<String> withMethod(String method) {
+    return msg -> msg.contains("\"method\":\"" + method + "\"");
```

Comment:
careful. This can fail if the JSON is formatted with a space between the `method` field and its value (it'd still be valid JSON and respecting to have `"method":  "aaa"`)
As such, consider parsing it as JSON.

Reply:
I wanted to avoid this for some reason, but I will have to do that